rancher/k3s
# Multiple small fixes

#### Proposed Changes ####

Multiple small 1/2 line fixes, one per commit:
1. Disable containerd experimental snapshot labels
    Related to #2455 and containerd/containerd#4684
    These were not meant to be enabled by default, break images with many layers, and will be disabled by default on the next containerd release.
1. Fix containerd sock path warning
    Resolves warning 1 from #2471
1. Add Cluster ID support to k3s stub cloud controller
    Resolves warning 2 from #2471.
    As per https://github.com/kubernetes/cloud-provider/issues/12 the ClusterID requirement was never really followed through on, so the flag is probably going to be removed in the future.
1. Disable configure-cloud-routes and external service/route programming support when using k3s stub cloud controller
    Resolves warning 3 from #2471
1. Respect --data-dir path for crictl.yaml
    Related to rancher/rke2#474
    Note that anyone who customizes the data-dir path will have to set CRI_CONFIG_FILE to the correct path when using the wrapped binaries (crictl, etc). This is better than dropping files in the incorrect location.
1. Fix size check and make script shell consistent
    Related to #2205

#### Types of Changes ####

* Bugfix

#### Verification ####

1. Pull image listed in https://github.com/rancher/k3s/issues/2455#issuecomment-720635256, confirm no error message.
1. Look for log message listed in #2471 part 1
1. Look for log message listed in #2471 part 2
1. Look for log message listed in #2471 part 3
1. Start k3s with --data-dir, confirm that no files are created in /var/lib/rancher/k3s
1. Build k3s or check CI logs, verify no errors while checking binary size.

#### Linked Issues ####

See issues linked above

#### Further Comments ####

One commit per issue, hopefully that's easier to review. I am not planning on squashing this before merging since the commits should all stand on their own.
